tracing-journald/src/lib.rs

```diff
@@ -39,7 +39,7 @@
 )]
 #[cfg(unix)]
 use std::os::unix::net::UnixDatagram;
-use std::{fmt, io, io::Write};
+use std::{fmt, io, io::Write, ops::ControlFlow};
```

Comment:
unfortunately, the `ControlFlow` type is not stable on our current MSRV (Rust 1.49): https://github.com/tokio-rs/tracing/runs/5620697371?check_suite_focus=true --- we would have to either bump our MSRV to 1.55, or change the function to return a `bool` or something to remain compatible with 1.49.

Reply:
i don't think we should make the MSRV conditional. i think we should either bump it so we can use `ControlFlow` in the API, or just use `bool` to avoid bumping it.

given that 1.55.0 _is_ permitted by our MSRV policy of stable + 3 versions, i think the bump would be acceptable

Reply:
(i would, however, wonder if using `ControlFlow` still makes things significantly clearer if we were to move the filtering functionality to another method with a name like `event_enabled`...)

Reply:
The one thing I worry about with a method like `event_enabled` is that it'd imply calling `event_enabled` and then conditionally calling `on_event`, which could imply duplicated work.

At the same time, though, splitting it would allow disabling the event callback for the entire layer stack (like regular enabled does) and not just layers lower in the stack, which could be a benefit.

(It's also worth noting that in my head I have a partially formed design for per-layer filtering which is basically `Tee(Layered): Subscribe`, which splits off the layer callbacks into a branch which can do whatever filtering/processing it wants using regular layer composition. I have no idea how practical it is, but it's better served by the layered form of enabling than the global form. But I also understand that the global form is important for optimization and disabling reporting of spans/events nobody is listening to.)

You all know the system better than I do, so I'll go in the direction you think is best.